### ISSUE TITLE
[dashboard properties] multiple issues around dashboard properties color labels 

### ISSUE THREAD

**Issue by junlincc** [apache/superset#11674]
<img width="1633" alt="Screen Shot 2020-11-12 at 1 02 41 PM" src="https://user-images.githubusercontent.com/67837651/98997228-097b7d80-24e9-11eb-8460-a5897a7a7059.png">

This line chart only has 5 lines^^
When exploring a chart from the dashboard, to preserve the colors in the dashboard all label colors are persisted in the chart metadata, causing lots of unnecessary color metadata in the payload

Expected behavior: all non-relevant labels should be dropped

**Comment by mistercrunch** [comment]
+1, let's rip this out. I think the intent was to offer color consistency while going from dashboards to explore, but it wasn't implemented the right way and creates a bunch of metadata in the wrong place.

**Comment by mistercrunch** [comment]
If the feature request is to have color consistency through the dashboard->explore flow, it should be implemented by passing a dashboard_id_for_context to explore, and explore can get the dashboard metadata it needs to guarantee that.

I think color consistency on that flow is nice, but not a huge priority given the complexity behind doing this nicely. 

Long term solution may be around the SPA (single page app) where explore may be a huge modal on top of the dashboard in some cases, and just have that context available in Redux.

**Comment by zuzana-vej** [comment]
I was just about to report related bug, which I think it covered in above comments (if needed let me know and I can open separate issue).
- User can select color scheme on chart. (This works as expected.)
- user can select color scheme on dashboard. (This works as expected.)
- If both color schemes are selected, dashboard color scheme takes priority over chart color scheme. (This works as expected.)
- When a user clicks on Explore chart on Dashboard and got into Chart Explore, same colors on that specific chart which user saw on dashboard should be passed to chart Explore. This used to work in the past as per @graceguo-supercat  but is **currently broken - Bug.**

**Comment by junlincc** [comment]
> * When a user clicks on Explore chart on Dashboard and got into Chart Explore, same colors on that specific chart which user saw on dashboard should be passed to chart Explore. This used to work in the past as per @graceguo-supercat  but is **currently broken - Bug.**

Yea i am aware of this bug😔 this entire area needs to be worked on. i am gonna change the issue title to more generic to cover all these issues. thanks for reporting as always!

**Comment by amitmiran137** [comment]
I think this was handled by @geido  lately
Can you confirm ?

**Comment by geido** [comment]
> I think this was handled by @geido  lately
> Can you confirm ?

Yes I worked on the color consistency recently. The label colors are in the payload to enable consistency with the Dashboard in the Explore view. However, the label colors are not automatically generated in a Dashboard as it used to happen before. The user must set them manually to override any given color. This should assure that the metadata will be most often very small. Also, this part is being reworked now for some additional requirements that we discussed lately. I'll keep this potential problem in mind. Thanks for the mention. Cc @stephenLYZ
